File: src/main/java/com/sairaj/blog/services/PostService.java
package com.sairaj.blog.services;

import java.util.List;

import org.springframework.stereotype.Service;

import com.sairaj.blog.entities.Category;
import com.sairaj.blog.entities.Post;
import com.sairaj.blog.payload.PostDto;
import com.sairaj.blog.payload.PostResponse;


public interface PostService {
	
	
	//create
	
	PostDto createPost(PostDto postDto, Integer userId, Integer categoryId);
	
	//update
	
	PostDto updatePost(PostDto postDto,Integer postId);
	
	//delete
	
	void deletePost(Integer postId);
	
	//get all posts
	
	PostResponse getAllPost(Integer pageNumber, Integer pageSize, String sortBy, String sortDir);
	
	//get single post
	
	PostDto getPostById(Integer postId);
	
	//get all post by category
	
	List<PostDto> getPostByCategory(Integer categoryId);
	
	//get post by user
	
	List<PostDto> getPostsByUser(Integer userId);
	
	//search post
	
	List<PostDto> searchPosts(String keyword);

}

File: Frontend/src/services/post-service.js
import { privateAxios } from "./helper";
import { myAxios } from "./helper";

export const createPost = (postData)=>{
    // console.log(postData)
    return privateAxios.post(`/api/user/${postData.userId}/category/${postData.categoryId}/posts`,postData).then((response)=> response.data)
}

//get all posts
export const getAllPosts = (pageNumber,pageSize) =>{
    return myAxios.get(`/api/posts?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=addedDate&sortDir=desc`).then(response=>response.data)
}

//load single post
export const loadPost=(postId)=>{
    return myAxios.get("/api/posts/" + postId).then((response) => response.data)
}

//create comment
export const createComment=(comment,postId)=>{
    return privateAxios.post(`/api/post/${postId}/comments`,comment)
}

//upload image
export const uploadPostImage=(image,postId)=>{
    let formData = new FormData()
    formData.append("image",image)

    return privateAxios.post(`/api/post/image/upload/${postId}`,formData).then((response)=>response.data)
}


//get category wise post
export const loadPostCategoryWise=(categoryId)=>{

    return privateAxios.get(`/api/category/${categoryId}/posts`).then(response=>response.data)
}

//load post user wise
export const loadPostUserWise=(userId)=>{
    return privateAxios.get(`/api/user/${userId}/posts`).then(response=>response.data)
}

//delete post
export const deletePostService=(postId)=>{
    return privateAxios.delete(`/api/posts/${postId}`).then(response=>response.data)
}

//update post
export const updatePostContent=(post,postId)=>{
    return privateAxios.put(`/api/posts/${postId}`,post).then(response=>response.data)
}